src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -467,5 +501,19 @@ public override T GetResult<T>()
                 return result.ToObject<T>(this.jsonSerializer);
             }
         }
+
+        private class JsonArrayPool<T> : IArrayPool<T>
```

Comment:
What is the purpose of this class? It seems to directly wrap `ArrayPool`.

Reply:
indeed. That's all it does. But as it implements newtonsoft's `IArrayPool<T>` interface, it allows us to use the .NET `ArrayPool<T>` type for the pool we offer to newtonsoft. The .NET class of course does not implement newtonsoft's interface directly.